component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -2519,4 +2528,115 @@ private long getUnlockTimeInMilliSeconds(AuthenticatedUser authenticatedUser) th
                     username, e);
         }
     }
+
+    /**
+     * Trigger event after generating Email OTP.
+     *
+     * @param request HttpServletRequest.
+     * @param context Authentication context.
+     * @throws AuthenticationFailedException Exception on authentication failure.
+     */
+    private void publishPostEmailOTPGeneratedEvent(HttpServletRequest request, AuthenticationContext context)
+            throws AuthenticationFailedException {
+
+        Map<String, Object> eventProperties = new HashMap<>();
+        eventProperties.put(IdentityEventConstants.EventProperty.CORRELATION_ID, context.getCallerSessionKey());
+        AuthenticatedUser authenticatedUser = (AuthenticatedUser) context.getProperty(EmailOTPAuthenticatorConstants
+                .AUTHENTICATED_USER);
+        eventProperties.put(IdentityEventConstants.EventProperty.USER_NAME, authenticatedUser.getUserName());
+        eventProperties.put(IdentityEventConstants.EventProperty.TENANT_DOMAIN, context.getTenantDomain());
+        eventProperties.put(IdentityEventConstants.EventProperty.USER_STORE_DOMAIN, authenticatedUser
+                .getUserStoreDomain());
+        eventProperties.put(IdentityEventConstants.EventProperty.APPLICATION_NAME, context.getServiceProviderName());
+        eventProperties.put(IdentityEventConstants.EventProperty.USER_AGENT, request.getHeader(
+                EmailOTPAuthenticatorConstants.USER_AGENT));
+        if (request.getParameter(EmailOTPAuthenticatorConstants.RESEND) != null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Setting true resend-code property in event since http request has resendCode parameter.");
+            }
+            eventProperties.put(IdentityEventConstants.EventProperty.RESEND_CODE,
+                    request.getParameter(EmailOTPAuthenticatorConstants.RESEND));
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("Setting false resend-code property in event since http request has not resendCode parameter.");
+            }
+            eventProperties.put(IdentityEventConstants.EventProperty.RESEND_CODE, false);
+        }
+
+        eventProperties.put(IdentityEventConstants.EventProperty.GENERATED_OTP, context.getProperty(
+                EmailOTPAuthenticatorConstants.OTP_TOKEN));
+
+
```

Comment:
unnecessary new line

Reply:
Fixed in 5a5d198b3054bf710058f1c945a1f73b350bd783